iotedgehubdev/certutils.py

```diff
@@ -227,16 +227,28 @@ def get_cert_file_path(id_str, dir_path):
     def get_pfx_file_path(id_str, dir_path):
         return os.path.join(dir_path, id_str, 'cert', id_str + EC.PFX_SUFFIX)
 
-    def export_cert_artifacts_to_dir(self, id_str, dir_path):
-        if Utils.check_if_directory_exists(dir_path) is False:
-            msg = 'Invalid export directory {0}'.format(dir_path)
+    def load_cert_from_file(self, id_str, cert_path, key_path, key_passphrase):
+        if id_str in list(self._cert_chain.keys()):
+            msg = 'Could not load cert from file. Certificate already in chain. ID: {0}'.format(id_str)
             raise EdgeValueError(msg)
+        cert_dict = {}
+        with open(cert_path, 'r') as cert_file:
+            cert_content = cert_file.read()
+            cert_dict['cert'] = crypto.load_certificate(crypto.FILETYPE_PEM, cert_content)
```

Comment:
load_certificate will not throw exception right?

Reply:
It will if something went wrong. Since there's no extra info to be added, no try catch is added here

Reply:
We don't let user know that the cert is invalid?

Reply:
If it's invalid, the crypto library will raise exception. CLI will fail and show the error from crypto library to user. Because there're various reasons to fail, the error from crypto could better explain what's going wrong

Reply:
We still can wrap the exception into EgeInvalidArgument exception. Thus if we want to custom the handle of this type of error outside, it will be better maintained. And this scenario is kind of invalid argument right?